daq/host.py

```diff
@@ -556,13 +582,28 @@ def _base_tests(self):
         self.record_result('base', state=MODE.DONE)
         return True
 
+    def _dhcp_port_toggle_test(self, logging_handler):
+        def ip_listener(target_ip):
+            LOGGER.info("%s test Received ip: %s" % (self.test_name, target_ip))
+            if logging_handler:
+                LOGGER.removeHandler(logging_handler)
+            self._ip_listener = None
+            self._end_test()
+
+        self.connect_port(False)
+        time.sleep(5)
+        self.connect_port(True)
+        self._ip_listener = ip_listener
+
     def _run_next_test(self):
         try:
             if self.remaining_tests:
                 LOGGER.debug('Target port %d executing tests %s',
                              self.target_port, self.remaining_tests)
                 self.timeout_handler = self._main_module_timeout_handler
                 self._docker_test(self.remaining_tests.pop(0))
+            elif self.dhcp_tests:
```

Comment:
I'm not sure exposing dhcp_tests is top-level construct is quite the right way to go here. I would have thought a "dhcp_test" would be a sub-class of a "inline_test" which is a subclass of a "module" (not a literal sub-class, but as a general concept). Going forward, there might be other kinds of inline tests that aren't DHCP, and there's going to be native tests that are modules but not docker tests.  Right now, there's a conflating of "docker_test" and "module" -- but that's just a collapsed abstraction and I'd move in the direction of pulling that apart.   I think the problem in the code here is actually the line above this (line 604), where "remaining_tests" maps 1:1 to "dhcp_tests", which is where I think it'd be better served to have the right abstraction in place.

If we don't know about the implementation, is there anything fundamentally different about the dhcp_tests from the other tests?